### PR TITLE
Moving unit tests out of IntegrationTests assembly 

### PR DESCRIPTION
**Pull request by bhsubra**
- Moved unit tests out of IntegrationTests assembly
- Created Microsoft.HttpRepl.Fakes and moved classes common from Microsoft.HttpRepl.Tests and Microsoft.HttpRepl.IntegrationTests

### CODE REVIEW

File: src/Microsoft.HttpRepl.Tests/Microsoft.HttpRepl.Tests.csproj [hunk 1]
```diff
@@ -6,11 +6,18 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
```
Comment (tlmii) [inline src/Microsoft.HttpRepl.Tests/Microsoft.HttpRepl.Tests.csproj]: Is this needed?

File: src/Microsoft.HttpRepl/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -1,3 +1,4 @@
 using System.Runtime.CompilerServices;
 
 [assembly: InternalsVisibleTo("Microsoft.HttpRepl.IntegrationTests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
```
Comment (tlmii) [inline src/Microsoft.HttpRepl/Properties/AssemblyInfo.cs]: Is this needed anymore?

Comment (bhsubra) [reply src/Microsoft.HttpRepl/Properties/AssemblyInfo.cs]: Yes, UriLauncher and String.cs are internal.

File: src/Microsoft.HttpRepl.Tests/Microsoft.HttpRepl.Tests.csproj (hunk 1 above)
Comment (bhsubra) [reply src/Microsoft.HttpRepl.Tests/Microsoft.HttpRepl.Tests.csproj]: Not required. Removed as part of this commit:
https://github.com/aspnet/HttpRepl/commit/e262cdca0ca93a629fd4dd674daaa799d3e199e5

File: src/Microsoft.HttpRepl/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (bhsubra) [reply src/Microsoft.HttpRepl/Properties/AssemblyInfo.cs]: As discussed offline, I was looking at the wrong one. Removed it here:
https://github.com/aspnet/HttpRepl/commit/04a5c4d9044b186a341072406e548882e1d539a4

Thanks!

**Comment by bhsubra** [comment]
@tlmii , as discussed offline, moved all the mocks under root directory in Microsoft.HttpRepl.Fakes assembly:
https://github.com/aspnet/HttpRepl/commit/b3e1c291c9f74c2af927c8ba7dbc373c89150723

**Pull request merged**